README.rst

```diff
@@ -2,55 +2,132 @@
-<https://libgeos.org/>`__ (the engine of `PostGIS
-<https://postgis.net/>`__) and `JTS
-<https://locationtech.github.io/jts/>`__ (from which GEOS is ported)
-libraries. Shapely is not concerned with data formats or coordinate systems,
-but can be readily integrated with packages that are. For more details, see:
-
-* `Shapely GitHub repository <https://github.com/shapely/shapely>`__
-* `Shapely documentation and manual <https://shapely.readthedocs.io/en/latest/>`__
+planar geometric objects. It is using the widely deployed open-source
+geometry library `GEOS <https://libgeos.org/>`__ (the engine of `PostGIS
+<https://postgis.net/>`__, and a port of `JTS <https://locationtech.github.io/jts/>`__).
+Shapely wraps the GEOS operations in NumPy ufuncs providing a performance
+improvement when operating on arrays of geometries, as well as provides a
+rich scalar `Geometry` interface.
+Shapely is not concerned with data formats or coordinate systems, but can be
+readily integrated with packages that are.
+
+What is a ufunc?
+----------------
+
+A universal function (or ufunc for short) is a function that operates on
+*n*-dimensional arrays in an element-by-element fashion, supporting array
+broadcasting. The for-loops that are involved are fully implemented in C
+diminishing the overhead of the Python interpreter.
+
+Multithreading
+--------------
+
+PyGEOS functions support multithreading. More specifically, the Global
+Interpreter Lock (GIL) is released during function execution. Normally in Python, the
+GIL prevents multiple threads from computing at the same time. PyGEOS functions
+internally releases this constraint so that the heavy lifting done by GEOS can be
+done in parallel, from a single Python process.
 
 Usage
 =====
 
 Here is the canonical example of building an approximately circular patch by
-buffering a point.
+buffering a point, using the scalar Geometry interface:
 
 .. code-block:: pycon
 
-    >>> from shapely.geometry import Point
-    >>> patch = Point(0.0, 0.0).buffer(10.0)
+    >>> import shapely
+    >>> patch = shapely.Point(0.0, 0.0).buffer(10.0)
     >>> patch
-    <shapely.geometry.polygon.Polygon object at 0x...>
+    <shapely.Polygon POLYGON ((10 0, 9.952 -0.98, 9.808 -1.951, 9.569 -2.903, 9....>
     >>> patch.area
-    313.65484905459385
+    313.6548490545941
+
+Using the vectorized ufunc interface, compare a grid of points with a polygon:
+
+.. code:: python
+
+    >>> geoms = shapely.points(*np.indices((4, 4)))
+    >>> polygon = shapely.box(0, 0, 2, 2)
+
+    >>> shapely.contains(polygon, geoms)
+    array([[False, False, False, False],
+           [False,  True, False, False],
+           [False, False, False, False],
+           [False, False, False, False]])
+
+Compute the area of all possible intersections of two lists of polygons:
 
-See the manual for more examples and guidance.
+.. code:: python
+
+    >>> polygons_x = shapely.box(range(5), 0, range(10, 15), 10)
```

Comment:
Likewise, these are a bit hard to visualize from the function calls.  Suggest using a smaller set of combinations, and printing them out.